Code_analyzer.cpp

```diff
@@ -0,0 +1,31 @@
+#include "Code_analyzer.h"
+
+void Code_analyzer::saveResult(string directory_path) const{
+
+    string result_info;
+
+    result_info.append("File: " + directory_path + '\n'
+        + "Physical lines count " + to_string(total_lines) + '\n'
+        + "Code lines count " + to_string(total_code_lines) + '\n'
+        + "Comment lines count " + to_string(total_blank_lines) + '\n'
+        + "Blank lines count " + to_string(total_comment_lines) + '\n');
+
+    ofstream output(directory_path + "Result.txt");
+    if (!output.is_open())
+        throw exception("can't open/create file");
+    output << result_info;
+
+    output.close();
+}
+void Code_analyzer::doAnalyze(string current_file) {
+
```

Comment:
Why do not you use any thread synchronization method? You use mutual variables from different threads.

Reply:
Resolved